app/models/signup_state.rb

```diff
@@ -18,9 +18,38 @@ class SignupState < ActiveRecord::Base
   scope :verified, -> { where(verified: true) }
   sifter :verified do verified.eq true end
 
+  def self.create_from_trusted_data(data)
+    role = User.roles[data[:role]] ? data[:role] : nil
+    SignupState.create!(
+      role: role,
+      verified: true,
+      contact_info_value: data[:email],
+      trusted_data: {
+        email: data[:email],
+        name:  data[:name],
+        uuid:  data[:external_user_uuid],
+        role:  role
+      }
+    )
+  end
+
+  def trusted?
+    trusted_data.present?
+  end
+
+  def skip_email_validation?
+    contact_info_value == trusted_data['email']
```

Comment:
Can we use existing `signup_state.verified?` in place of `skip_email_validation?`

Reply:
Actually what we need is a method to determine what should be rendered after the first role / email entry screen.  For normal users, it's email verification.  For verified instructors it's password, and for verified students it's direct to profile.  

Another wrinkle is that accounts are created when the password is, so for verified students we have to create the account first.

I added a `trusted_student` controller action/handler to accomplish this.